addons/mail_plugin/controllers/mail_plugin.py

```diff
@@ -53,6 +53,59 @@ def res_partner_enrich_and_create_company(self, partner_id):
 
         return response
 
+    @http.route('/mail_plugin/partner/enrich_and_update_company', type='json', auth='outlook', cors='*')
+    def res_partner_enrich_and_update_company(self, partner_id):
+        """
+        Enriches an existing company using IAP
+        """
+        partner = request.env['res.partner'].browse(partner_id)
+        if not partner.is_company:
+            return {'error': _('Contact must be a company')}
+
+        normalized_email = partner.email_normalized
+        if not normalized_email:
+            return {'error': _('Contact has no valid email')}
+
+        domain = tools.email_domain_extract(normalized_email)
+        iap_data = self._iap_enrich(domain)
+
+        if 'enrichment_info' in iap_data:
+            return {'enrichment_info': iap_data['enrichment_info']}
+
+        phone_numbers = iap_data.get('phone_numbers')
+
+        fields_mapping = {'street': 'street_name', 'city': 'city', 'zip': 'postal_code', 'website': 'domain'}
+
+        to_write = {db_key: iap_data.get(iap_key) for db_key, iap_key in fields_mapping.items() if not getattr(partner, db_key)}
```

Comment:
I would move this after the `if not partner.image_128:` to organize the code a bit better.

And sugg:
```py
model_fields_to_iap_mapping = {
    'street': 'street_name',
    'city': 'city',
    'zip': 'postal_code',
    'website': 'domain'
}

# only update keys for which we don't have a value yet
partner_values = {
    model_field: iap_data.get(iap_key)
    for model_field, iap_key in model_fields_to_iap_mapping.items() if not partner[model_field]
}
```

More readable?